core/src/main/resources/lib/form/checkbox.jelly

```diff
@@ -41,10 +41,10 @@ THE SOFTWARE.
       If this attribute is unspecified or null, it defaults to unchecked, otherwise checked.
     </st:attribute>
     <st:attribute name="id" />
-    <st:attribute name="onclick" />
+    <st:attribute name="onclick" deprecated="true" />
```

Comment:
Would it make sense to add a description of what to do instead? Even just a URL to this PR or something might be useful to developers, e.g. "Avoid inline JS, see https://github.com/jenkinsci/jenkins/pull/6852 for an example of what to do instead."